### PR TITLE
Do not `resolveIndexDtype` in lowering

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Instead, make our system capable of handling `DataType::Index`. `resolveIndexDtype` is a workaround for us for not having a well handling of different integer types. I am fixing this issue right now in https://github.com/NVIDIA/Fuser/pull/696, therefore, this workaround should be removed as well.

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!build

**Comment by zasdfgbnm** [comment]
!build

File: csrc/device_lower/lower2device.cpp [hunk 1]
```diff
@@ -270,21 +270,15 @@ void GpuLower::lower(Fusion* fusion) {
...
   // Copy fusion into a new kernel for processing
-  kernel_ = std::make_unique<kir::Kernel>(fusion, kernel_index_type);
+  kernel_ = std::make_unique<kir::Kernel>(fusion, indexType());
   // Alias the fusion kernel caries around as a view of itself.
   fusion_ = kernel_.get();
 
-  // Convert tensor views of DataType::Index type to either Int or Int32
-  for (auto tv : ir_utils::allTvs(fusion_)) {
-    if (tv->dtype() == DataType::Index) {
-      tv->resolveIndexDtype();
```
Comment (naoyam) [inline csrc/device_lower/lower2device.cpp:285]: Should we remove `Val::resolveIndexDtype` as well?

**Review by naoyam — APPROVED** [review]
LGTM. Just to confirm, with this change, tensors with type `DataType::Index` will be generated as `Tensor<nvfuser_index_t, ...>`?

File: csrc/device_lower/lower2device.cpp (hunk 1 above)
Comment (zasdfgbnm) [reply csrc/device_lower/lower2device.cpp:285]: It is currently still used here: https://github.com/NVIDIA/Fuser/blob/e5ef1a22990bd0292efaea9381b713d6f2d3f0ab/csrc/device_lower/pass/scalar_hoist.cpp#L464
we should eventually remove this, but we need first to make sure we are using `DataType::Index` where it should be.

**Pull request merged**